lib/kaur/result_tuple.ex

```diff
@@ -38,9 +60,18 @@ defmodule Kaur.ResultTuple do
     {:error, "oops"}
   """
   @spec sequence([result_tuple]) :: ({:ok, [any()]}|{:error, any()})
-  def sequence(list), do: do_sequence(list, [])
+  def sequence(list) do
+    case Enum.reduce_while(list, [], &reducer/2) do
+      {:error, _} = err -> err
+      result -> {:ok, Enum.reverse result}
+    end
+  end
+
+  defp reducer(elt, acc) do
```

Comment:
* I don't really like `reducer` as a function name as it's really bound to the `sequence` function?
* `elt` => `element`?
* `acc` => `elements`, `valid_elements`?